ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/assertions/CommandDeduplicationAssertions.scala

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure.assertions
+
+import java.time.Instant
+
+import com.daml.api.util.DurationConversion
+import com.daml.ledger.api.testtool.infrastructure.participant.CompletionResponse
+import com.daml.ledger.api.v1.completion.Completion
+import com.daml.ledger.api.v1.experimental_features.CommandDeduplicationPeriodSupport.{
+  DurationSupport,
+  OffsetSupport,
+}
+import com.daml.lf.data.Ref
+import com.daml.scalautil.Statement.discard
+import com.google.protobuf.duration.Duration
+
+object CommandDeduplicationAssertions {
+  def assertDeduplicationDuration(
+      requestedDeduplicationDuration: Duration,
+      previousSubmissionSendTime: Instant,
+      completionReceiveTime: Instant,
+      completion: Completion,
+      durationSupport: DurationSupport,
+  ): Unit = discard {
+    val requestedDuration = DurationConversion.fromProto(requestedDeduplicationDuration)
+    durationSupport match {
+      case DurationSupport.DURATION_NATIVE_SUPPORT =>
+        completion.deduplicationPeriod.deduplicationDuration.map { reportedDurationProto =>
```

Comment:
If you use `forEach` in these cases, then you probably don't need `discard` anymore.

Reply:
Thanks, we're gonna fail here.